### PR TITLE
feat: survey form layout

### PR DESCRIPTION
**Pull request by luke-mcfarlane-rocketlab**
## JIRA Ticket

[BSS-382](https://jira.csiro.au/browse/BSS-382)

## Description

Changed the layout of the create site record form to be in a single long view.

## Proposed Changes

- updated form component to show list with headings rather than tabs
- moved errors to the end of the form

## How to Test

This was tested using the Rapid Impact Survey.

1. Navigate to a survey
2. Add a new record
3. Verify layout and form functionality
4. (optional) Open an already created record and verify layout and functionality

## Additional Information

This is an update rather than configurable. Happy to discuss the option of being configurable but it seems currently there are a few bugs in the form code and if we were to maintain both we will need to address these in each version of the component.

## Checklist

- [x] I have confirmed all commits have been signed.
- [x] I have added JSDoc style comments to any new functions or classes.
- [x] Relevant documentation such as READMEs, guides, and class comments are updated.

### CODE REVIEW

**Review by stevecassidy — COMMENTED** [review]
The layout looks ok but we need to have a configuration option to select between this and the old layout.

**Comment by stevecassidy** [comment]
Thinking about this, the option to use inline vs tabs for sections should be a configuration on the form in the uiSpec.  Then I could if I wanted have one form with tabs and another with inline sections.  Need to add in a configuration option to the Designer.

**Comment by luke-mcfarlane-rocketlab** [comment]
@stevecassidy issues above have been addressed

**Review by stevecassidy — CHANGES_REQUESTED** [review]
Generally ok, see my small comments.  We need to document this change to the notebook format but I'll make a separate ticket for that and include the other recent change.

File: app/src/gui/components/record/form.tsx [hunk 1]
```diff
@@ -1185,29 +1167,169 @@ class RecordForm extends React.Component<
...
+                                fontWeight: 'bold',
+                                padding: '10px',
+                              }}
+                            >
+                              {ui_specification.views[view].label}
+                            </div>
+                            <Form>
+                              {description !== '' && (
+                                <Box
+                                  bgcolor={'#fafafa'}
```
Comment (ranisa-gupta16) [inline app/src/gui/components/record/form.tsx:1207]: Can you use theme specific color variables?

**Comment by luke-mcfarlane-rocketlab** [comment]
@stevecassidy @ranisa-gupta16 comments actioned

**Pull request merged**